PMM-13899 pull upstream changes v0.17.1

BupycHuk opened the pull request:
PMM-13899

https://github.com/Percona-Lab/pmm-submodules/pull/3876

prombot committed:
Update common Prometheus files (#951)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#963)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

shamil committed:
pg_replication_slot: add slot type label (#960)

Signed-off-by: Alex Simenduev <shamil.si@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.44.0 to 0.45.0 (#948)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.44.0 to 0.45.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Commits](https://github.com/prometheus/common/compare/v0.44.0...v0.45.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model (#949)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.4.1-0.20230718164431-9a2bf3000d16 to 0.5.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/commits/v0.5.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sitole committed:
pg_stat_database: added support for `active_time` counter (#961)

* feat(pg_stat_database): active time metric

---------

Signed-off-by: Jiri Sveceny <jiri.sveceny@icloud.com>

dependabot[bot] committed:
Bump golang.org/x/crypto from 0.14.0 to 0.17.0 (#988)

Bumps [golang.org/x/crypto](https://github.com/golang/crypto) from 0.14.0 to 0.17.0.
- [Commits](https://github.com/golang/crypto/compare/v0.14.0...v0.17.0)

---
updated-dependencies:
- dependency-name: golang.org/x/crypto
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.17.0 to 1.18.0 (#993)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.17.0 to 1.18.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.17.0...v1.18.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

keegancsmith committed:
use Info level for excluded databases log message (#1003)

This is the only log message which didn't specify a level in the
postgres_exporter. I am unsure if this log message should be info or
debug, but leaning towards the more important since previously it would
just always log.

The way I validated this was the only non-leveled logger was via grep.
Both of these only returned this callsite previously:

  git grep 'logger\.Log'
  git grep '\.Log(' | grep -v level

Signed-off-by: Keegan Carruthers-Smith <keegan.csmith@gmail.com>

jocelynthode committed:
Add connection limits metrics for pg_roles and pg_database (#997)

* Add database connection limits metrics

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

* Add roles connection limits metrics

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

* Fix copyright year

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Fix spacing in pgDatabaseQuery

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Fix case on pgRolesConnectionLimitsQuery

Co-authored-by: Joe Adams <github@joeadams.io>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>

* Do not add roleMetrics when row is not valid

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>

---------

Signed-off-by: Jocelyn Thode <jocelyn@thode.email>
Signed-off-by: Jocelyn Thode <jocelynthode@users.noreply.github.com>
Co-authored-by: Joe Adams <github@joeadams.io>

dependabot[bot] committed:
Bump github.com/DATA-DOG/go-sqlmock from 1.5.0 to 1.5.2 (#1000)

Bumps [github.com/DATA-DOG/go-sqlmock](https://github.com/DATA-DOG/go-sqlmock) from 1.5.0 to 1.5.2.
- [Release notes](https://github.com/DATA-DOG/go-sqlmock/releases)
- [Commits](https://github.com/DATA-DOG/go-sqlmock/compare/v1.5.0...v1.5.2)

---
updated-dependencies:
- dependency-name: github.com/DATA-DOG/go-sqlmock
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.18.0 to 1.19.0 (#1011)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.18.0 to 1.19.0.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/v1.19.0/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.18.0...v1.19.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_model from 0.5.0 to 0.6.0 (#1010)

Bumps [github.com/prometheus/client_model](https://github.com/prometheus/client_model) from 0.5.0 to 0.6.0.
- [Release notes](https://github.com/prometheus/client_model/releases)
- [Commits](https://github.com/prometheus/client_model/compare/v0.5.0...v0.6.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_model
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump google.golang.org/protobuf from 1.32.0 to 1.33.0 (#1014)

Bumps google.golang.org/protobuf from 1.32.0 to 1.33.0.

---
updated-dependencies:
- dependency-name: google.golang.org/protobuf
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.10.0 to 0.11.0 (#992)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.10.0 to 0.11.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.10.0...v0.11.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump golang.org/x/net from 0.20.0 to 0.23.0 (#1021)

Bumps [golang.org/x/net](https://github.com/golang/net) from 0.20.0 to 0.23.0.
- [Commits](https://github.com/golang/net/compare/v0.20.0...v0.23.0)

---
updated-dependencies:
- dependency-name: golang.org/x/net
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

MarcWort committed:
feat: Add safe_wal_size and wal_status to replication_slot  (#1027)

* feat: Add safe_wal_size to replication_slot

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

* feat: Add wal_status to replication_slot

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

---------

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

MarcWort committed:
fix: Only query active_time on pg>=14 (#1045)

Signed-off-by: MarcWort <113890636+MarcWort@users.noreply.github.com>

fhackenberger committed:
Update README.md (#1038)

Better example for the quick start with prometheus config and avoiding deprecated env variables.

Signed-off-by: fhackenberger <florian@hackenberger.at>

saz committed:
stop logging errors on replicas, fixes #547 (#1048)

Signed-off-by: Steffen Zieger <github@saz.sh>

prombot committed:
Update common Prometheus files (#983)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1076)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

tjhop committed:
chore!: adopt log/slog, drop go-kit/log (#1073)

* ci: update go to version 1.23

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

* build(deps): bump prometheus/{client_golang,common,exporter-toolkit}

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

* chore!: adopt log/slog, drop go-kit/log

The bulk of this change set was automated by the following script which
is being used to aid in converting the various exporters/projects to use
slog:

https://gist.github.com/tjhop/49f96fb7ebbe55b12deee0b0312d8434

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>

---------

Signed-off-by: TJ Hoplock <t.hoplock@gmail.com>
Co-authored-by: Ben Kochie <superq@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/client_golang from 1.20.4 to 1.20.5 (#1079)

Bumps [github.com/prometheus/client_golang](https://github.com/prometheus/client_golang) from 1.20.4 to 1.20.5.
- [Release notes](https://github.com/prometheus/client_golang/releases)
- [Changelog](https://github.com/prometheus/client_golang/blob/main/CHANGELOG.md)
- [Commits](https://github.com/prometheus/client_golang/compare/v1.20.4...v1.20.5)

---
updated-dependencies:
- dependency-name: github.com/prometheus/client_golang
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.60.0 to 0.60.1 (#1080)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.60.0 to 0.60.1.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.60.0...v0.60.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

bitfehler committed:
Make walreceiver collector useful w/o repmgr (#1086)

In a streaming replication setup that was created without replication
manager (`repmgr`), the `stat_wal_receiver` collector does not return
any metrics, because one value it wants to export is not present.

This is rather overly opinionated. The missing metric is comparatively
uninteresting and does not justify discarding all the others. And
replication setups created without `repmgr` are not exactly rare.

This commit makes the one relevant metric optional and simply skips it
if the respective value cannot be determined.

Signed-off-by: Conrad Hoffmann <ch@bitfehler.net>

prombot committed:
Update common Prometheus files (#1083)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.0 to 0.13.1 (#1081)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.0 to 0.13.1.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.0...v0.13.1)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1087)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

sysadmind committed:
Update changelog and version for a v0.16.0 release (#1088)

Signed-off-by: Joe Adams <github@joeadams.io>

sysadmind committed:
Fix version header in changelog (#1089)

Signed-off-by: Joe Adams <github@joeadams.io>

jyothikirant-sayukth committed:
Update pg_long_running_transactions.go (#1092)

To extract time in seconds for pg_long_running_transactions_oldest_timestamp_seconds query which currently return epoch time.

Signed-off-by: Jyothi Kiran Thammana <147131742+jyothikirant-sayukth@users.noreply.github.com>

aagarwalla-fx committed:
Fix to replace dashes with underscore in the metric names (#1103)

* Fix to replace dashes with underscore in the metric names

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

* Code style fix

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

---------

Signed-off-by: aagarwalla-fx <arpit.agarwalla@falconx.io>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.1 to 0.13.2 (#1108)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.1 to 0.13.2.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.1...v0.13.2)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

n-rodriguez committed:
Checkpoint related columns in PG 17 have been moved from pg_stat_bgwriter to pg_stat_checkpointer (#1072)

* Checkpoint related columns in PG 17 have been moved from pg_stat_bgwriter to pg_stat_checkpointer

Fix https://github.com/prometheus-community/postgres_exporter/issues/1060

See: https://www.dbi-services.com/blog/postgresql-17-new-catalog-view-pg_stat_checkpointer/
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Add support for pg_stat_checkpointer

See: https://www.dbi-services.com/blog/postgresql-17-new-catalog-view-pg_stat_checkpointer/
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Run integration tests with Postgres 17

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

* Update date in file header

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

---------

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

prombot committed:
Update common Prometheus files (#1090)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

prombot committed:
Update common Prometheus files (#1110)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

khiemdoan committed:
Add Postgres 17 for CI test (#1105)

Signed-off-by: Khiem Doan <doankhiem.crazy@gmail.com>

dependabot[bot] committed:
Bump github.com/prometheus/common from 0.61.0 to 0.62.0 (#1118)

Bumps [github.com/prometheus/common](https://github.com/prometheus/common) from 0.61.0 to 0.62.0.
- [Release notes](https://github.com/prometheus/common/releases)
- [Changelog](https://github.com/prometheus/common/blob/main/RELEASE.md)
- [Commits](https://github.com/prometheus/common/compare/v0.61.0...v0.62.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/common
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

prombot committed:
Update common Prometheus files (#1124)

Signed-off-by: prombot <prometheus-team@googlegroups.com>

NevermindZ4 committed:
pg_stat_statements PG17 (#1114)

Signed-off-by: Nevermind <79126473+NevermindZ4@users.noreply.github.com>

michael-todorovic committed:
fix: handle pg_replication_slots on pg<13 (#1098)

* fix: handle pg_replication_slots on pg<13

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: tests

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

---------

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

fgalind1 committed:
feat: add wait/backend to pg_stat_activity (#1106)

Signed-off-by: Felipe Galindo Sanchez <felipe.galindo.sanchez@intel.com>

bitfehler committed:
Export last replay age in replication collector (#1085)

The exported replication lag does not handle all failure modes, and can
report 0 for replicas that are out of sync and incapable of recovery.

A proper replacement for that metric would require a different approach
(see e.g. #1007), but for a lot of folks, simply exporting the age of
the last replay can provide a pretty strong signal for something being
amiss.

I think this solution might be preferable to #977, though the lag
metric needs to be fixed or abandoned eventually.

Signed-off-by: Conrad Hoffmann <ch@bitfehler.net>

dependabot[bot] committed:
Bump github.com/prometheus/exporter-toolkit from 0.13.2 to 0.14.0 (#1126)

Bumps [github.com/prometheus/exporter-toolkit](https://github.com/prometheus/exporter-toolkit) from 0.13.2 to 0.14.0.
- [Release notes](https://github.com/prometheus/exporter-toolkit/releases)
- [Changelog](https://github.com/prometheus/exporter-toolkit/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prometheus/exporter-toolkit/compare/v0.13.2...v0.14.0)

---
updated-dependencies:
- dependency-name: github.com/prometheus/exporter-toolkit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

sysadmind committed:
Fix missing dsn sanitization for logging (#1104)

This log line was not sanitized previously which could result in logging sensitive information. I have scanned the rest of the files and I don't see anywhere else that DSN is used in a log line without this filter.

Resolves #1042

Signed-off-by: Joe Adams <github@joeadams.io>

n-rodriguez committed:
Skip pg_stat_checkpointer collector if pg<17 (#1112)

* fix: skip collector if pg<17

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: better condition

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>

* fix: fix PGStatCheckpointerCollector tests

Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>

---------

Signed-off-by: Michael Todorovic <michael.todorovic@outlook.com>
Signed-off-by: Nicolas Rodriguez <nico@nicoladmin.fr>
Co-authored-by: Michael Todorovic <michael.todorovic@outlook.com>

sysadmind committed:
Prep for v0.17 (#1127)

Signed-off-by: Joe Adams <github@joeadams.io>

vancwo committed:
Fix: Handle incoming labels with invalid UTF-8 (#1131)

It's possible that incoming labels will contain invalid UTF-8 characters. This results in a panic. This fix sanitizes the label's string to ensure only valid UTF-8 characters are included, by replacing invalid characters with � (REPLACEMENT CHARACTER)

Signed-off-by: Cooper Worobetz <cooper@worobetz.ca>

SuperQ committed:
Release v0.17.1 (#1132)

* [BUGFIX] Fix: Handle incoming labels with invalid UTF-8 #1131

Signed-off-by: SuperQ <superq@gmail.com>

BupycHuk committed:
Merge tag 'v0.17.1' into PMM-7-pull-upstream-changes-v0.17.1

v0.17.1

# -----BEGIN PGP SIGNATURE-----
#
# iQIzBAABCgAdFiEEoFiTUofnXYMugU5GxkayPJ4yRfEFAme/HEkACgkQxkayPJ4y
# RfFouRAAgSwPJ3YTvi4CeNvgrRTQrFsYK8NF7evyGPUZi2nWQm9XP+eW3zQt0Pxr
# AMw2JkibUow7KGGuzAbEY07FyK+yY3JgMBf9+g1nymMEGiT+rA/cvLZnEU5CrP8P
# VdLxKfB1kz/iVHKSmAo7WKF0TP1Dso/Oma56J594ACpxarI2UVQtBGKhk8LUvcGH
# 1Yd0ji2FtezU3M17vZAoZru0ZcS1bbyxP5u6tz6+zqrxHbtFp+aMRAoghc1bafO0
# OgbXvVr8qqKwVvvD2Nt6kDo7iCazQINp5FsMGVKuFyMLxLv9ohaVBV1Lwx11+Ts1
# fRqMz5Co1dYUusic68CnOWRZ8iJrZysi/Jxv/lG1U/5fgIfYMu2CzlgfNZnWTAYw
# 1IRKoMBsmg6zg67Xq959pyB9HgultpjMHp0pVOasFJa40QWNuYp0V8B0SpHzZQ1Q
# gckQBByFHDKM1XGtM6MMbCamKU38MiYKJ8OtIZ/I4jlM7bA0IINvZt2o5ZMWl3OO
# 7Bx2V/MxeEyowrVmb80NeAjm+FMrMFNpW2mGUnzwMmTXdkv5kvZLW3wBx71tYSnF
# O3AD227WnosnjGepVU5iBAIoEP+tI6Pm6D80jPHvlOCl5T+hp+f4y7BnTEnizmz2
# yVKzhiOVutkFQHa2tDw9WC+eeKLD7A7z718R2PLyQARweTwuZ1Y=
# =umzG
# -----END PGP SIGNATURE-----
# gpg: Signature made Wed Feb 26 16:51:05 2025 +03
# gpg:                using RSA key A058935287E75D832E814E46C646B23C9E3245F1
# gpg: Can't check signature: No public key

# Conflicts:
#	.github/workflows/golangci-lint.yml
#	cmd/postgres_exporter/datasource.go
#	cmd/postgres_exporter/main.go
#	cmd/postgres_exporter/namespace.go
#	cmd/postgres_exporter/postgres_exporter.go
#	cmd/postgres_exporter/probe.go
#	collector/pg_replication_slot.go
#	collector/pg_replication_slot_test.go
#	collector/pg_stat_bgwriter.go
#	collector/pg_stat_bgwriter_test.go
#	collector/probe.go
#	go.mod
#	go.sum

BupycHuk committed:
PMM-7 pull upstream changes from v0.17.1

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/not_signed)](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=311) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/postgres_exporter?pullRequest=311) before we can accept your contribution.<br/>**1** out of **21** committers have signed the CLA.<br/><br/>:white_check_mark: BupycHuk<br/>:x: shamil<br/>:x: sitole<br/>:x: keegancsmith<br/>:x: jocelynthode<br/>:x: MarcWort<br/>:x: fhackenberger<br/>:x: saz<br/>:x: bitfehler<br/>:x: sysadmind<br/>:x: jyothikirant-sayukth<br/>:x: aagarwalla-fx<br/>:x: n-rodriguez<br/>:x: NevermindZ4<br/>:x: michael-todorovic<br/>:x: fgalind1<br/>:x: vancwo<br/>:x: SuperQ<br/>:x: khiemdoan<br/>:x: tjhop<br/>:x: prombot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/postgres_exporter?pullRequest=311) it.</sub>

BupycHuk commented on collector/pg_replication_slot.go:
```diff
@@ -76,12 +94,31 @@ var (
 	FROM pg_replication_slots;`
+	pgReplicationSlotNewQuery = `SELECT
+		slot_name,
+		slot_type,
```
```suggestion
		plugin,
		slot_type,
```

BupycHuk committed:
PMM-7 Fix build

BupycHuk committed:
PMM-7 Fix build

BupycHuk committed:
PMM-7 Remove unused dependencies from go.mod and go.sum

Cleaned up `go.mod` and `go.sum` by removing `github.com/go-kit/log` and `github.com/go-logfmt/logfmt`, which are no longer used in the project. This helps keep the dependency list minimal and maintains a cleaner project state.

ademidoff commented on .github/workflows/container_description.yml:
```diff
@@ -0,0 +1,57 @@
+  PushDockerHubReadme:
+    runs-on: ubuntu-latest
+    name: Push README to Docker Hub
+    if: github.repository_owner == 'prometheus' || github.repository_owner == 'prometheus-community' # Don't run this workflow on forks.
```
That action will never execute with such an `if` statement, so why not remove it?

BupycHuk replied:
to not have conflicts when we pull new version.

ademidoff reviewed (commented):
Looking at the test failures, I believe there should be a way to make the tests pass. WDYT?

BupycHuk committed:
PMM-7 Add "plugin" label to pg_replication_slot metrics

This change introduces the "plugin" label to pg_replication_slot metrics, providing more detailed information about the replication slot. The updated metrics now include the plugin name alongside existing labels, improving observability and filtering capabilities.

ademidoff commented:
Shall we have a feature build for this update?

BupycHuk commented:
attached link to FB
